pages/register.js

```diff
@@ -1,151 +1,129 @@
 import { useForm } from 'react-hook-form'
 import { useRouter } from 'next/router'
-import { useRef } from 'react'
-import Image from 'next/image'
-import Navbar from '../components/layout/Navbar'
-import Head from 'next/head'
+import axios from 'axios'
+import React, { useRef } from 'react'
 
 export default function Register () {
   const { register, handleSubmit, watch, formState: { errors } } = useForm()
   const password = useRef({})
-  //  const router = useRouter()
+  const router = useRouter()
+  const { redirect } = router.query
   password.current = watch('password', '')
 
-  const registerUser = async (data) => {
-    const response = await fetch('/api/register', {
-      body: JSON.stringify({
-        companyName: data.companyName,
-        fullName: data.fullName,
-        email: data.email,
-        password: data.password
-      }),
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      method: 'POST'
-    })
-    const result = await response.json()
-    if (result.status === 302) {
-      message = 'User already exits'
-      //  router.push('/home')
-    } else {
-      //  router.push('/register')
+  const registerUser = async ({ companyName, fullName, email, password }) => {
+    try {
+      const { data } = await axios.post('/api/register', {
+        companyName,
+        fullName,
+        email,
+        password
+      })
+      if (data) {
+        alert('User created successfully.')
+      }
+      router.push(redirect || '/')
+    } catch (error) {
+      if (error.response.status === 406) {
+        alert('User already exists.')
+      } else if (error.response.status === 417) {
```

Comment:
here you can do just an `else` when you change the api code to the code I suggested in the review

Reply:
Done